### DISCUSSION TITLE
CSSTree doesn't throw errors on invalid pseudo classes?

### DISCUSSION THREAD

**Discussion by u-RraaLL** [uBlockOrigin/uBlock-issues#2440] [Q&A] [answered]
Before CSSTree, if I made a mistake with a pseudo class  - e.g. a typo, the whole line would be invalidated. Now, the editors will accept anything that doesn't match the name of a procedural filter or an action operator <- these will throw errors until properly finished e.g. `:remove` -> `:remove()`, `:matches-path` -> `:matches-path(/^/$/)`.

This, however, will not throw any errors in "My filters" `:matchess-path(/^/$/)`. It will in Picker, though, which, I'm guessing, verifies the presence of brackets better.

Invalid pseudo classes don't throw errors in neither of the places. They just won't have a match, which in Picker's case will usually prevent creation of the filter.

At first I thought Picker would accept `##a, b:randomstringhere`, allowing for matches or `a`, but it seems it looks at the invalid pseudo classes in the procedural manner `##a, b:has(c)`, not finding a match for either.

It will still accept them if placed inside `:is()`, `:where()` or `:matches()` though. Which makes sense since they're using forgiving selectors list. So I understand the errors won't be thrown for these, but I do think they should be thrown in the other cases.

---
I probably should've filed it as a bug report, but I'd be forced to lie on the template to send it in.

**Comment by ghost** [comment, marked as answer]
https://github.com/uBlockOrigin/uBlock-issues/issues/2442

**Comment by krystian3w** [comment]
I think share something to Github mobile app still ignore templates.
